src/gpl/src/initialPlace.cpp

```diff
@@ -41,6 +43,40 @@
 #include "graphics.h"
 
 #include "utl/Logger.h"
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include "cusparse.h"
+#include "cusolverDn.h"
+#include "cusolverSp.h"
+
+#define CUDA_ERROR(value) do { \
+   cudaError_t m_cudaStat = value; \
+   if(m_cudaStat != cudaSuccess){ \
+       fprintf(stderr, "Error %s at line %d in file %s\n", cudaGetErrorString(m_cudaStat), __LINE__, __FILE__); \
+       exit(-1); \
+   } \
+} while(0)
+#define CUSPARSE_ERROR(value) do { \
```

Comment:
Avoid macros

Reply:
This is what I always do in CUDA coding. Without macros, it is hard to detect the status of CUDA or CUSPARSE. CUDA's principle is: The code never stopped even if it has an error. It just returns a wrong status.  Maybe I have limited knowledge. I will bring this point and talk with Chris about this.